magma_access_gateway_installer/agw_network_configurator.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+import os
+from subprocess import check_call, check_output
+
+import ipcalc  # type: ignore[import]
+import yaml
+from systemd.journal import JournalHandler  # type: ignore[import]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(JournalHandler())
+logger.setLevel(logging.INFO)
+
+
+class AGWInstallerNetworkConfigurator:
+
+    DNS_ADDRESSES = "8.8.8.8 208.67.222.222"
+    INTERFACES_DIR = "/etc/network/interfaces.d"
+    REBOOT_NEEDED = False
+
+    def __init__(self, network_interfaces, eth0_address, eth0_gw_ip_address):
+        self.eth0_address = eth0_address
+        self.eth0_gw_ip_address = eth0_gw_ip_address
+        self.network_interfaces = network_interfaces
+
+    def configure_network_interfaces(self):
+        """Configures network interfaces if necessary."""
+        if not self._network_interfaces_are_named_eth0_and_eth1:
+            self._update_interfaces_names_in_cloud_init()
+            self._configure_grub()
+            self._update_grub_cfg()
+        if not self._dns_configured:
+            self._configure_dns()
+        if not self._network_interfaces_config_files_exist:
+            self._create_interfaces_config_files()
+        if self._netplan_installed:
+            self._remove_netplan()
+        if not self._networking_service_enabled:
+            self._enable_networking_service()
+
+    @property
+    def _network_interfaces_are_named_eth0_and_eth1(self) -> bool:
+        """Checks whether available network interfaces are named correctly."""
+        return all(interface in self.network_interfaces for interface in ["eth0", "eth1"])
+
+    @staticmethod
+    def _configure_grub():
+        """Adds required configuration to /etc/default/grub"""
+        logger.info("Configuring /etc/default/grub...")
+        original_grub_file = open("/etc/default/grub", "r")
+        original_grub_file_content = original_grub_file.readlines()
+        updated_grub_file_content = [
+            line.replace(line, 'GRUB_CMDLINE_LINUX="net.ifnames=0 biosdevname=0"\n')
+            if line.startswith("GRUB_CMDLINE_LINUX=")
+            else line
+            for line in original_grub_file_content
+        ]
+        original_grub_file.close()
+
+        updated_grub_file = open("/etc/default/grub", "w")
+        updated_grub_file.writelines(updated_grub_file_content)
```

Comment:
We could use `with` like in the `_update_interfaces_names_in_cloud_init` method

Reply:
Done.